nucypher/network/nodes.py

```diff
@@ -759,30 +776,31 @@ def learn_from_teacher_node(self, eager=False):
                 self.log.info(f"Verification Failed - "
                               f"Cannot establish connection to {sprout}.")
 
-            except sprout.StampNotSigned:
-                self.log.warn(f'Verification Failed - '
-                              f'{sprout} stamp is unsigned.')
-
-            except sprout.NotStaking:
-                self.log.warn(f'Verification Failed - '
-                              f'{sprout} has no active stakes in the current period '
-                              f'({self.staking_agent.get_current_period()}')
-
-            except sprout.InvalidWorkerSignature:
-                self.log.warn(f'Verification Failed - '
-                              f'{sprout} has an invalid wallet signature for {sprout.decentralized_identity_evidence}')
-
-            except sprout.UnbondedWorker:
-                self.log.warn(f'Verification Failed - '
-                              f'{sprout} is not bonded to a Staker.')
-
-            except sprout.Invalidsprout:
-                self.log.warn(sprout.invalid_metadata_message.format(sprout))
-
-            except sprout.SuspiciousActivity:
-                message = f"Suspicious Activity: Discovered sprout with bad signature: {sprout}." \
-                          f"Propagated by: {current_teacher}"
-                self.log.warn(message)
+            # TODO: This whole section is weird; sprouts down have any of these things.
```

Comment:
What do you think needs to happen with the exception handling here - is it duplicated?

Reply:
I honestly don't know.  I've intentionally kinda left it out of my mind for the moment in an effort to stay focused on this PR; it needs to be addressed in a later one, along with a generalized high-level effort toward optimal logging and error handing.